### PR TITLE
Workflow: Run Lerna directly from GitHub action when publishing to npm targeting WP core

### LINKED ISSUE #53628: Packages: Publishing to npm fails for `wp/X.Y` branches because of older Node.js
### Description

Additional context in https://github.com/WordPress/gutenberg/pull/53565. While trying to manually publish WordPress packages to npm using the script executed from the local machine, I run into an error where `npm ci` failed because `npm` version set in `trunk` to be able to run the script doesn't meet the requirements for the branch where source code lives. In effect, the script bails out, and publishing isn't possible.

In effect, any backport to WordPress 6.3 and lower will fail from GitHub UI as explained:

https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/release.md#wordpress-releases

Everything should work correctly for WordPress 6.4, as the script and the future branch will use the same Node.js and npm versions.

### Step-by-step reproduction instructions

I executed from `trunk` that is now on Node 16 and npm 8:

```bash
./bin/plugin/cli.js npm-latest --semver minor
```

It pulled changes from `release/16.4` that was still on Node 14 and npm 6, so `npm ci` throws an error that it can finish when using npm 8.

The same issue will happen when trying to backport changes to WordPress core when using any of `wp/X.Y` branches that enforce usage of npm 6.

### Screenshots, screen recording, code snippet

_No response_

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by gziolo**
## What?

Fixes #53628.

> Additional context in https://github.com/WordPress/gutenberg/pull/53565. While trying to manually publish WordPress packages to npm using the script executed from the local machine, I run into an error where npm ci failed because npm version set in trunk to be able to run the script doesn't meet the requirements for the branch where source code lives. In effect, the script bails out, and publishing isn't possible.
> 
> In effect, any backport to WordPress 6.3 and lower will fail from GitHub UI as explained:
> 
> https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/release.md#wordpress-releases
> 
> Everything should work correctly for WordPress 6.4, as the script and the future branch will use the same Node.js and npm versions.

## Why?

This is exactly what has happened when trying to publish npm packages targeting WordPress 6.3.1 as documented in https://github.com/WordPress/gutenberg/pull/53623. For reference the job that failed:

https://github.com/WordPress/gutenberg/actions/runs/5874702589/job/15929869480#step:8:409

Screenshot 2023-08-17 at 15 17 26

Th script got executed with Node.js 16 and npm 8, but the `wp/6.3` branch runs on Node.js 14 and npm 6. In addition to that, the branch has strict setting allowing only npm 6.

## How?

I ended up simplifying the way we publish when targeting WordPress core releases. We run Lerna directly, as it turns out we never update changelog files, nor cherry-pick commits to `trunk` as in other scenarios.

I also removed all npm scripts that we stopped using:

- `npm run publish:check` 
- `npm run publish` 
- `npm run publish:patch` 

I updated documentation to use Lerna commands directly.

## Testing Instructions

I don't know what would be the best way to test it without merging. It should just work the next time we have to publish npm packages for WordPress 6.4.x minor release following the instructions:

https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/release.md#wordpress-releases

### CODE REVIEW

File: docs/contributors/code/release.md [hunk 1]
```diff
@@ -4,41 +4,41 @@ The [Gutenberg repository](https://github.com/WordPress/gutenberg) on GitHub is
 
 Before you begin, there are some requirements that must be met in order to successfully release a stable version of the Gutenberg plugin. You will need to:
 
-- Be a member of the [Gutenberg development team](https://developer.wordpress.org/block-editor/block-editor/contributors/repository-management/#teams). This gives you the ability to launch the GitHub actions that are related to the release process and to backport pull requests (PRs) to the release branch.
-- Have write permissions on the [Make WordPress Core](http://make.wordpress.org/core) blog. This allows you to draft the release post.
-- Obtain approval from a member of the Gutenberg Core team in order to upload the new version Gutenberg to the WordPress.org plugin directory.
+-   Be a member of the [Gutenberg development team](https://developer.wordpress.org/block-editor/block-editor/contributors/repository-management/#teams). This gives you the ability to launch the GitHub actions that are related to the release process and to backport pull requests (PRs) to the release branch.
```
Comment (priethor) [inline docs/contributors/code/release.md:7]: There are many extra blank spaces between the markdown hyphen and the text. Are those expected? Do they change anything? It doesn't seem to affect the processed markdown result.

Comment (gziolo) [reply docs/contributors/code/release.md:7]: It's the formatting that Prettier enforces. I have it enabled in my IDE, and it looks like we don't format markdown files in `docs` folder. It happens automatically for README files for packages, for instance. I think @Mamaduka applied the same set of changes in https://github.com/WordPress/gutenberg/commit/578ad3936c05e7699dd2464db7310df38bfdd36b. It's also why I need to rebase this PR, and most likely, most of the whitespace changes will disappear.

Comment (gziolo) [reply docs/contributors/code/release.md:7]: I still see many changes and I confirmed it's coming from Prettier by executing:

Screenshot 2023-08-18 at 11 15 11

https://github.com/WordPress/gutenberg/pull/53762/commits/21a8d1c32cc9ca61fb275e2ab954e1104f5a5438 reflects the opinions that Prettier has.

Comment (priethor) [reply docs/contributors/code/release.md:7]: Thanks for confirming!

**Review by priethor — APPROVED** [review]
Changes make sense to me. Given there is no way to fully test this until the next package release, I'm all in for trying ✅

**Comment by gziolo** [comment]
> Changes make sense to me. Given there is no way to fully test this until the next package release, I'm all in for trying ✅

I discussed it with @tellthemachines, and she plans to test this revised workflow on Monday. I will help with any alternatives in case anything goes wrong.

**Comment by gziolo** [comment]
Unfortunately, the changes aren't enough, and the workflow for the `wp` type is broken. It misses the part from the Node.js script it used before that runs git checkout on `wp/6.3` :upside_down_face: I triggered npm publishing with `wp` type and `6.3` version: https://github.com/WordPress/gutenberg/actions/runs/5915550747. However, it performed a patch release from `trunk` with `wp-6.3` dist tag. I will fix it manually. I reverted the commit manually in `trunk` with 540192a61219b5422d2b57b76a1f5ecd62f0bda6. I will set proper versions for packages set for `wp-6.3` dist-tag.

**Pull request merged**